libflux/src/flux/ast/mod.rs

```diff
@@ -319,6 +333,10 @@ impl BaseNode {
     pub fn is_empty(&self) -> bool {
         self.errors.is_empty() && !self.location.is_valid()
     }
+
+    pub fn add_comments(&mut self, comments: CommentList) {
+        self.comments = comments;
```

Comment:
I noticed comment list is a singly linked list. Does this method need to check if `self.comments` already has comments? The name `add_comments` implies it might already have comments and we are adding new ones. If this is really a replace/set operation should we rename the function to `set_comments`? Also does the function need to be public?

Reply:
The name `set_comments` is definitely more appropriate. I updated that. About public ... good question ... just verified that it needs to be public because BaseNode is defined in the `ast` module, but comments are populated in `parser` module.